elyra/pipeline/registry.py

```diff
@@ -0,0 +1,483 @@
+        super().__init__()
+
+    def parse_component_details(self, component_body, component_name):
+        # TODO: Maybe use this loop to get description?
+        for line in component_body:
+            print(line)
+
+        label = ' '.join(component_name.split('_')).title()
+        description = ""
+
+        component_json = {
+            'label': label,
+            'image': "",
+            'id': get_id_from_name(label),
+            'description': ' '.join(description.split()),
+            'node_types': []
+        }
+
+        node_type = set_node_type_data(get_id_from_name(label), label, description)
+        component_json['node_types'].append(node_type)
+
+        return component_json
+
+    def parse_component_properties(self, component):
+        return None
+
+
+class ComponentReader(SingletonConfigurable):
+    _type = 'local'
+
+    def __init__(self):
+        super().__init__()
+
+    def get_component_body(self, component_path):
+        raise NotImplementedError()
+
+
+class FilesystemComponentReader(ComponentReader):
+    _type = 'file'
+    _dir_path: str = 'resources'
+
+    def __init__(self):
+        super().__init__()
+
+    def get_component_body(self, component_path):
+        component_dir = os.path.join(os.path.dirname(__file__), self._dir_path)
+        component_file = os.path.join(component_dir, component_path)
+
+        component_extension = os.path.splitext(component_path)[-1]
+        with open(component_file, 'r') as f:
+            if component_extension == '.yaml':
+                try:
+                    return yaml.safe_load(f), None
+                except yaml.YAMLError as e:
+                    raise RuntimeError from e
+            elif component_extension == '.py':
+                return f.readlines()
+            else:
+                raise ValueError(f'File type {component_extension} is not supported.')
+
+
+class UrlComponentReader(ComponentReader):
+    _type = 'url'
+    _url_path: str
+
+    def __init__(self):
+        super().__init__()
+
+    def get_component_body(self, component_path):
+        parsed_path = urllib.parse.urlparse(component_path).path
+        component_name, component_extension = os.path.splitext(parsed_path)
+        component_body = urllib.request.urlopen(component_path)
+
+        if component_extension == ".yaml":
+            try:
+                return yaml.safe_load(component_body), None
+            except yaml.YAMLError as e:
+                raise RuntimeError from e
+        elif component_extension == ".py":
+            return component_body.readlines(), component_name.split('/')[-1]
+        else:
+            raise ValueError(f'File type {component_extension} is not supported.')
+
+
+class ComponentRegistry(SingletonConfigurable):
+
+    parsers = {
+        KfpComponentParser._type: KfpComponentParser(),
+        AirflowComponentParser._type: AirflowComponentParser(),
+        ComponentParser._type: ComponentParser()
```

Comment:
This doesn't seem conducive to _bring your own processors_, but something we can deal with when/if that time comes.

Reply:
Yeah this is definitely one of those things where I wasn't sure exactly how it might affect future changes, but I'll keep in for now I guess

Reply:
I think, like runtime pipeline processors, these "component parsers" will need to be registered via entry points as well.  Or perhaps, be a function of the runtime processor implementation.  I.e., the `RuntimePipelineProcessor` subclass provides "its" ComponentParser implementation.  That way third parties just register the runtime processor.  This topic goes with [this discussion](https://github.com/elyra-ai/elyra/discussions/1672).  That said, this is somewhat orthogonal to getting the general support framework in place - so nothing to worry about now.
@lresende - thoughts?